distributor/src/distributor/subscription_filter.bal

```diff
@@ -0,0 +1,87 @@
+import ballerina/auth;
+import ballerina/encoding;
+import ballerina/http;
+import ballerina/log;
+import ballerina/websub;
+
+const HUB_TOPIC = "hub.topic";
+const HUB_CALLBACK = "hub.callback";
+
+map<string> callbackMap = {};
+
+public type SubscriptionFilter object {
+    *http:RequestFilter;
+
+    public function filterRequest(http:Caller caller, http:Request request, http:FilterContext context) returns boolean {
+        map<string>|error params = request.getFormParams();
+
+        if params is error {
+            log:printError("error extracting form params", params);
+            return false;
+        }
+
+        map<string> paramMap = <map<string>> params;
+        if !paramMap.hasKey(HUB_TOPIC) || !paramMap.hasKey(HUB_CALLBACK) {
+            log:printError("topic and/or callback not available");
+            return false;
+        }
+
+        string topic = paramMap.get(HUB_TOPIC);
+        string callback = paramMap.get(HUB_CALLBACK);
+        
+        string|error decodedTopic = encoding:decodeUriComponent(topic, "UTF-8");
+        if (decodedTopic is string) {
+            topic = decodedTopic;
+        } else {
+            log:printWarn("error decoding topic, using the original form: " + topic + ". Error: " + decodedTopic.toString());
+        }
+
+        if (topic != JSON_RESULTS_TOPIC) {
```

Comment:
We need to support image sending too .. so user has to subscribe to both the JSON and the image. 

See: https://github.com/ECLK/Results-Dist/issues/48

Reply:
We can fix this after merging this code.